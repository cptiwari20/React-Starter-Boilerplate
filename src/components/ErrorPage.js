import React from 'react';
import { Link } from 'react-router-dom';

const ErrorPage = () => (
  <div>
    <h1>Sorry! No page found.</h1>
    <p>Go to Dashboard.</p>
    <Link to='/'>Home</Link>
  </div>
);

export default ErrorPage;