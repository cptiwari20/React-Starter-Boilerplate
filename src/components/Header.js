import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startSignOut } from '../actions/auth';

export const Header = (props) => (
  <header className="header">
    <div className='content-container'>
      <div className='header__container'>
        <Link to='/' 
          className="header__title"
        >
            <h1>MyReactApp</h1>
        </Link>
       <div>
        <button 
          className='button button--link'
          onClick={props.startSignOut}>
          Log Out
        </button>
       </div>
      </div>
    </div>
  </header>
);

export default connect(null, { startSignOut })(Header);