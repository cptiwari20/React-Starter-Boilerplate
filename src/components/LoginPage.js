import React from 'react';
import { connect } from 'react-redux';
import { startSignIn } from '../actions/auth';

export const LoginPage = (props) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">MyReactApp</h1>
      <p>Write app subtitle here.</p>
      <button 
        className='button'
        onClick={props.startSignIn}> 
        Login With Google 
      </button>
    </div>
  </div>
);

export default connect(null, { startSignIn })(LoginPage);