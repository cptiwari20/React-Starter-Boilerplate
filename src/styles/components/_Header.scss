.header{
  background: $dark-grey;
};

.header__container{
  align-items: center;
  display: flex;
  justify-content: space-between;
  padding: $s-size 0;
}

.header__link{
  text-decoration: none;
  color: white;
  font-weight: 300;
  padding: 0 $l-size;
}

.header__title{
  color: white;
  text-decoration: none;
  h1{
    margin: 0;
  }
}