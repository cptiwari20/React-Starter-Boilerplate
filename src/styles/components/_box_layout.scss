.box-layout{
  align-items: center;
  background: url('/images/bg.jpg');
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;

};

.box-layout__box{
  background: fade-out(white, 0.17);
  border-radius: 10px;
  padding: $l-size $s-size;
  text-align: center;
  width: 23rem;
};

.box-layout__title{
  margin: 0 0 $m-size 0;
  line-height: 1;
}
