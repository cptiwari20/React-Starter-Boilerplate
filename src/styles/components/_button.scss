.button{
  background: $dark-grey;
  border-radius: 10px;
  border: none;
  color: white;
  display: inline-block;
  font-size: 1.8rem;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
};

.button--link{
  background: none;
};

.button--secondary{
  background: #888888;
}