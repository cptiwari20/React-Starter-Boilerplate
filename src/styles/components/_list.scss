.list-header{
  background: $off-white;
  border: 1px solid $off-white;
  color: $grey;
  display: flex;
  justify-content: space-between;
  padding: $m-size;
};

.list-item{
  border: 1px solid $off-white;
  color: $dark-grey;
  display: flex;
  flex-direction: column;
  margin: $s-size 0;
  padding: $s-size;
  text-decoration: none;
  transition: background .3s ease;
  word-wrap: break-all;
  &:hover{
    background: $off-white;
  };

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    align-content: center;
    justify-content: space-between;
    padding: $m-size;
  }
};

.list-body{
  padding-bottom: $m-size;
  
  @media (min-width: $desktop-breakpoint) {
    padding-bottom: $l-size;
  }
}

.list-item__message{
  padding: $m-size;
  border: 1px solid $off-white;
  display: flex;
  color: $grey;
  justify-content: space-around;

}

.list-item__title{
  margin: 0;
  word-break: break-all;
};

.list-item__subtitle{
  color: $grey;
  font-size: $m-size;
  word-break: break-all;
}