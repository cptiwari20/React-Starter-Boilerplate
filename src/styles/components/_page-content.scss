.page-header{
  background: #f5f5f5;
  margin-bottom: $l-size;
  padding: $l-size 0;
};

.page-header__title{
  margin: 0;
  font-weight: 300;
  padding-bottom: $l-size;
  span {
    font-weight: bold;
  }
}