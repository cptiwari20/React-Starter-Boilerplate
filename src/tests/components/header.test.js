import React from "react";
import { shallow } from 'enzyme';
import { Header } from "../../components/Header";


let wrapper, startSignOut;
beforeEach(()=> {
  startSignOut = jest.fn();
  wrapper = shallow(<Header startSignOut={startSignOut} />);
})

it('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

it('should call signOut Action on Button Click', () => {
  wrapper.find('button').simulate('click');

  expect(startSignOut).toBeCalled();
})
