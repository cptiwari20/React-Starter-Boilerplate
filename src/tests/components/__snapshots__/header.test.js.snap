// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__container"
    >
      <Link
        className="header__title"
        replace={false}
        to="/"
      >
        <h1>
          MyReactApp
        </h1>
      </Link>
      <div>
        <button
          className="button button--link"
          onClick={[MockFunction]}
        >
          Log Out
        </button>
      </div>
    </div>
  </div>
</header>
`;
